bugfix: #3344 Reconnecting to the same SFU on membership change

BillCarsonFr opened the pull request:
Fixes #3344

~~This a an ugly workaround, the root cause is that we have a problem with react hooks and a re-render is causing effects with non properly working dependencies to be ran again?~~
This has changed to actually address the root cause: the `activeFocus` state returned by the `useActiveLivekitFocus` hook was updating with new objects that had the same values. (so we switched to the same focus)
Now `useActiveLivekitFocus` behaves as expected, only returning new objects when there is a change.

BillCarsonFr committed:
bugfix: #3344 Reconnecting to the same SFU on membership change

BillCarsonFr committed:
fixup! commit error

toger5 commented on src/livekit/openIDSFU.ts:
```diff
@@ -55,13 +61,28 @@ export function useOpenIDSFU(
+      currentFocus.current = {} as LivekitFocus;
       setSFUConfig(undefined);
     }
   }, [client, activeFocus, showErrorBoundary]);
```
Using the primitive types in the dependecy array, should also make this work:
```suggestion
  }, [client, activeFocus?.type, activeFocus?.livekit_service_url, activeFocus?.livekit_alias, showErrorBoundary]);
```
We might need to then recompose the active focus to make eslint happy:
```
getSFUConfigWithOpenID(client,
    {
      type: activeFocus.type,
      livekit_service_url: activeFocus.livekit_service_url,
      livekit_alias: activeFocus.livekit_alias
      }
     ).then(...
```
or:
```
// inside the effect
const activeFocus = {
      type: activeFocus.type,
      livekit_service_url: activeFocus.livekit_service_url,
      livekit_alias: activeFocus.livekit_alias
}
```
But we dont need the ref and the comparison function anymore.

robintown committed:
Keep useActiveLivekitFocus from changing focus spuriously

robintown committed:
Remove redundant fix for spurious focus changes

We've now fixed it at the source by prohibiting state changes in useActiveLivekitFocus itself.
